From 18a59393ef5f14c3cfd7cac1662459426ceef41d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 9 Jan 2021 02:09:44 +0100
Subject: [PATCH 1/1] tools: efivar.py should check GUID when deleting

When deleting a variable we must check that the GUID provided by the
user matches the GUID of the variable.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 tools/efivar.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/efivar.py b/tools/efivar.py
index a02b09d46a..29eb90a235 100755
--- a/tools/efivar.py
+++ b/tools/efivar.py
@@ -149,7 +149,7 @@ class EfiVariableStore:
         offs = 0
         while offs < len(self.ents):
             var, loffs = self._next_var(offs)
-            if var.name == name and str(var.guid):
+            if var.name == name and str(var.guid) == guid:
                 if var.attrs != attrs:
                     print("err: attributes don't match")
                     exit(1)
-- 
2.29.2

